### ISSUE TITLE
Harvest Doctype

### ISSUE THREAD

**Issue by agritheory** [ERPNext-Agri-Community/erpnext-agri#3]
## Problem statement
Basically this is a document bridging the quantities harvested, datetime stamp, explicit UOM conversions, and item mapping (to cover primary crops and by products)

A persistent/ stateful harvest record should exist for each harvest, so that 1) data points can be created for management over time and 2) trace-ability to crop cycle can be enforced with a submit-able doctype

## Challenges
Harvest never matches expected harvest, both larger and smaller, byproducts and primary crops

## Possible solution to Problem
Make a harvest doctype
Create a harvest from Crop Cycle and from the harvest task inside crop cycle

## Reference screenshots / mockups (if any)
Mockup: 
![image](https://user-images.githubusercontent.com/18033666/41264022-6a56cdb6-6db8-11e8-8b25-259cf88ea494.png)
https://github.com/agritheory/erpnext/commit/3c757e5d82b14ee309aa434015b60ae3f598cea1

## Checklist :
- [ ] Finalized on solution (out of several)
- [ ] Finalized on solution implementation (of one)
- [ ] Implemented solution (as code)
- [ ] Made PR (link to PR here please)

**Comment by Alchez** [comment]
More details on the Harvest doctype can be found [here](https://share.nuclino.com/p/Harvest-Design-Meeting-with-Tyler-Krthx1yfmAMBqrjZnSdtqK).

**Comment by agritheory** [comment]
> More details on the Harvest doctype can be found here.
Action items synthesized from Alain's (as always) detail documentation:

- Need a plan for completing harvests. I propose a `percent_complete` field, with the logic behind it modified by the "Annual/ Perennial/ Biennial" field in Crop. I think this is more straight forward than the harvest by area or harvest by amount. I think for the user's sake we should have one way to do this "correctly", even if there are multiple valid approaches. 
- More alignment of terminology: "Continuous" vs "Annual, etc". I think we're going to need to build a ERPNext Ag-specific glossary with all the terms that people might want to use to describe something

**Comment by Tropicalrambler** [comment]
I agree Tyler, your clear thinking should prevail in this matter. Keep it simple first, then we can always complicate it later if enough people ask for it

**Comment by agritheory** [comment]
Convo with @Tropicalrambler 
Generalize harvest to seriliazed and unserialized crops and animals by leveraging a dynamic link. Other doctypes need to be added: Animal Group (A container for serialized or unserialized animals, replaces "Herd" in my livestock app and Plant (for a serialized Plant). 
![image](https://user-images.githubusercontent.com/18033666/41323841-0d9063f4-6e7f-11e8-8a61-ab397408224a.png)

**Comment by agritheory** [comment]
Also discussed a Life Cycle or Life Plan document for livestock that mirrors Crop/ Crop Plan. 
Crop Cycle and Animal Group both need a flag for "Is serialized"

**Comment by Tropicalrambler** [comment]
I propose this structure for the Child Table:

|    Stock Item        | Harvest UOM |   Yield   | Destination Warehouse | 
| -----------------|:-------------:|:--------:|:-----------------------:|
| Red Clover Seed  | Pound               | 150.00   | Barn Bin ABC1             |

For Livestock:
Can be done against a **Life Cycle** or **herd, brood, etc.**

|    Stock Item        | Harvest UOM |   Yield   | Destination Warehouse | 
| -----------------|:-------------:|:--------:|:-----------------------:|
| Medium Eggs      | Unit                | 150.00   | Egg cleaning station     |

**Comment by Tropicalrambler** [comment]
As I was working on the Fertilizer Application or Pesticide Application, it dawned on me that we can simplify greatly by having a **SINGLE** DocType which handles all the functionality for these types of work:
1. Planting
2. Transplanting / moving animals
3. Doing field work on the plants/ field / animals
4. Applying Fertilizer / feeding
5. Applying Pesticides / medication
6. Harvesting
7. Post-harvest work

# Work Done
Each of the above should belong to a **Work Done** DocType, and upon opening, the farmer selects:
a. Crop Cycle or Life Cycle (or herd)
b. Specific Crop Cycle or Life Cycle
c. Work Type (as above)
d. Date

The selections made in these four fields, enable or disable those child tables relevant to the **type** of work being done.  A harvest does not involve the application of pesticides.

**Comment by Tropicalrambler** [comment]
More about this here:
https://app.nuclino.com/ERPNext-Agri/General/AgriNext-Pending-Issues-and-Focus-d1a21188-19de-4057-9c46-3737064e227e
